crawl-ref/source/MSVC/stdint.h

```diff
@@ -125,7 +125,7 @@ typedef uint64_t  uintmax_t;
 
 
 // 7.18.2 Limits of specified-width integer types
-
+#define __STDC_LIMIT_MACROS
```

Comment:
Did you intend to leave this in? If so, the next line should be changed, and probably a comment should be added explaining the purpose of this line.

Reply:
Yep! I changed it to match https://github.com/crawl/crawl/pull/878/files#diff-13a5ee50343a26462bbfe00974d4c914 ; this file wasn't changed in the past because it wasn't used by anything. I'm using it now for projects I haven't pushed yet.

There's no harm in changing it, since it's only used by MSVC. I don't know why it works, but without the line, chrono and ratio throw hundreds of errors.

I'd rather not mess with the remainder of the file, since this bugfix is the limit of my experience with it.

See 
https://github.com/ftylitak/qzxing/issues/54
https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/3a3dc941-af59-4788-b75e-74e585a25f3e/compile-issue-vs2015-undeclared-its-in-stdinth-intmaxmax?forum=vcgeneral

Reply:
My point is that it's silly to have code that looks like:

    #define x
    #if defined(x) 
    // body...
    #endif

The entire body is now included unconditionally and all the preprocessor stuff is just noise. 

But it looks like this file wasn't really written as part of crawl at all, but is basically part of the standard library - I would be quite suspicious of making any change at all to stdint.h. Rather, we should work with its existing interface, which is to define `__STDC_LIMIT_MACROS` before including this file, to customise its behaviour. My understanding of the article you linked is that it suggests that the "right" fix is to include `<cstdint>` instead of including `<stdint.h>` directly - cstdint will take care of defining this flag for us, if necessary, before including stdint.h itself. Have you tried making that change to all the DCSS source files, and found that it does not fix the problem for MSVC?

Reply:
Completely agree.

Changing it to <cstdint> didn't work, but explicitly defining __STDC_LIMIT_MACROS in the preprocessor list did; while effectively the same thing, it's definitely the better location for the explicit definition.